ilc/server/registry/factory.js

```diff
@@ -1,9 +1,13 @@
 const config = require('config');
 const localStorage = require('../../common/localStorage');
 const Registry = require('./Registry');
-const wrapFetchWithCache = require('../../common/wrapWithCache');
+const CacheWrapper = require('../../common/CacheWrapper');
 const reportingPluginManager = require('../plugins/reportingPlugin');
-
+const { context } = require('../context/context');
 const logger = reportingPluginManager.getLogger();
 
-module.exports = new Registry(config.get('registry.address'), wrapFetchWithCache(localStorage, logger), logger);
+module.exports = new Registry(
+    config.get('registry.address'),
+    new CacheWrapper(localStorage, logger, context.getStore()),
```

Comment:
I might be wrong but it seems that `getStore` has to be called inside the call stack, because it will dump async context, and if we do it on the module level, it always will be the same...

Reply:
Updated